### PR TITLE
Fix merkle wiped on updated claim conditions

### PR DESCRIPTION
**Pull request by atbe**
Fixes issue #242 

This bug **only effects the drop module**. It does not effect the Bundle Drop module because the Bundle Drop modules *appends* new root entries to the `merkle` dictionary, while the Drop module does an entire replace.

### CODE REVIEW

File: src/modules/drop.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ import {
   ethers,
 } from "ethers";
 import { JsonConvert } from "json2typescript";
+import { C } from "ts-toolbelt";
```
Comment (jakeloo) [inline src/modules/drop.ts]: is this used?

File: src/modules/drop.ts [hunk 2]
```diff
@@ -594,13 +594,22 @@ export class DropModule
...
-    if (factory.allSnapshots().length === 0 && "merkle" in metadata) {
-      metadata["merkle"] = {};
-    } else {
+
+    const defaultMerkleRoot = hexZeroPad([0], 32).toString();
+
+    if (factory.allSnapshots().length > 0) {
       metadata["merkle"] = merkleInfo;
+    } else if (
+      factory.allSnapshots().length === 0 &&
```
Comment (jakeloo) [inline src/modules/drop.ts:607]: this condition is always true :D

**Review by joaquim-verges — APPROVED** [review]
lg - i'll copy over your new tests in v2 to make sure i'm good there

File: src/modules/drop.ts [hunk 3]
```diff
@@ -594,13 +594,22 @@ export class DropModule
...
+      conditions
+        .filter((c) => c.merkleRoot !== defaultMerkleRoot)
+        .map((c) => c.merkleRoot).length === 0
```
Comment (jakeloo) [inline src/modules/drop.ts:610]: probably don't need `.map`

**Review by jakeloo — CHANGES_REQUESTED** [review]

**Review by jakeloo — APPROVED** [review]

File: src/modules/drop.ts (hunk 1 above)
Comment (atbe) [reply src/modules/drop.ts]: been removed

File: src/modules/drop.ts (hunk 2 above)
Comment (atbe) [reply src/modules/drop.ts:607]: why do you think that?

File: src/modules/drop.ts (hunk 3 above)
Comment (atbe) [reply src/modules/drop.ts:610]: good point

File: src/modules/drop.ts (hunk 2 above)
Comment (atbe) [reply src/modules/drop.ts:607]: oh good point, the `if` condition right above 🤦🏽‍♂️

Comment (atbe) [reply src/modules/drop.ts]: fixed

**Pull request merged**